spec/lib/stellar/client_spec.rb

```diff
@@ -136,6 +137,52 @@
         end
         expect(native_asset_balance_info["balance"].to_f).to eq 250.0
       end
+
+      it "sends a native payment to the account through a channel account", vcr: {record: :once, match_requests_on: [:method]} do
+        client.create_account(
+          funder: source,
+          account: destination,
+          starting_balance: 100,
+        )
+
+        client.create_account(
+          funder: source,
+          account: channel_account,
+          starting_balance: 100,
+        )
+
+        amount = Stellar::Amount.new(150)
+
+        tx = client.send_payment(
+          from: channel_account,
+          to: destination,
+          amount: amount,
+          source_account: source
+        )
+        tx_hash = tx._attributes
+                    .instance_variable_get(:@collection)[:hash]
+
+        operation = client.horizon
+                          .transaction(hash: tx_hash)
+                          .operations
+                          ._get
+
+        operation_from_address = operation.records
+                                          .first
+                                          ._attributes
+                                          .instance_variable_get(:@collection)["from"]
+
+        source_account_info = client.account_info(source)
```

Comment:
I don't think you need to fetch the source account into separately, since we already know that (you generated the account above).

Try:

```ruby
expect(operation_from_address).to eq channel_account.address
```

Reply:
oh thanks